.github/workflows/ci.yml

```diff
@@ -428,9 +429,71 @@ jobs:
           if compgen -G './logs-${{ matrix.mapdl-version }}/*.log' > /dev/null ;then for f in ./logs-${{ matrix.mapdl-version }}/*.log; do echo "::group:: Log file $f" && cat $f && echo "::endgroup::" ; done; fi
           if compgen -G './logs-${{ matrix.mapdl-version }}/*.out' > /dev/null ;then for f in ./logs-${{ matrix.mapdl-version }}/*.out; do echo "::group:: Output file $f" && cat $f && echo "::endgroup::" ; done; fi
 
+  build_test_ubuntu:
+    name: Local Unit Testing on Ubuntu
+    runs-on: ubuntu-latest
+    container:
+      image: ghcr.io/pyansys/mapdl:v22.2-ubuntu
+      options: "--entrypoint /bin/bash"
+      credentials:
+        username: ${{ secrets.GH_USERNAME }}
+        password: ${{ secrets.GITHUB_TOKEN }}
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Setup Python
+        uses: actions/setup-python@v4.3.0
+        with:
+          python-version: 3.9
+
+      - name: Checking python
+        run: |
+          python -m pip install --upgrade pip
+
+      - name: Install OS packages
+        run: |
+          apt update
+          apt install -y libgl1-mesa-glx xvfb
+
+      - name: Test virtual framebuffer
+        run: |
+          python -m pip install -r .ci/requirements_test_xvfb.txt
+          xvfb-run python .ci/display_test.py
+
+      - name: Install ansys-mapdl-core
+        run: |
+          python -m pip install build
+          python -m build
+          python -m pip install dist/*.whl
+          xvfb-run python -c "from ansys.mapdl import core as pymapdl; print(pymapdl.Report())"
+
+
+      - name: Unit Testing Requirements Installation
+        run: |
+          python -m pip install -r requirements/requirements_tests.txt
+
+      # - name: DPF Server Activation
+      #   run: |
+      #     docker pull ghcr.io/pyansys/dpf-core:22.2dev
+      #     docker run -d --name dpfserver -p ${{ env.DPF_PORT }}:50052 ghcr.io/pyansys/dpf-core:22.2dev && echo "DPF Server active on port ${{ env.DPF_PORT }}."
+
+      - name: Unit Testing
+        run: |
+          unset PYMAPDL_PORT
+          unset PYMAPDL_START_INSTANCE
+          export ANSYSLMD_LICENSE_FILE=1055@${{ secrets.LICENSE_SERVER }}
+          export AWP_ROOT222=/ansys_inc
+          xvfb-run pytest -v -k "not test_database and not test_dpf" --durations=10 --maxfail=10  --reruns 7 --reruns-delay 3 --only-rerun MapdlExitedError  --cov=ansys.mapdl.core --cov-report=xml --cov-report=html
```

Comment:
Only comment, and not a blocker, is that we could mark tests with `pytest.mark.dpf` and you could deselect (or select) those and it might be easier.